ITU-BDSA2024-GROUP6/Chirp
# # feat(testing): Implement basic API tests

This commit introduces basic API testing capabilities to the Chirp project. It sets up a new xUnit test project specifically for API tests, allowing us to verify the functionality of our public endpoints.

Changes include:
- Create new Chirp.ApiTests project in the 'test' directory
- Implement ApiTests.cs with two initial test cases:
  1. Verify the public cheeps endpoint returns OK status
  2. Verify a user's timeline endpoint returns OK status
- Update project structure to accommodate the new test project
- Add necessary NuGet packages for xUnit testing

To run the tests:
1. Start the Chirp server: `dotnet run` in the Chirp.Razor directory
2. Run tests: `dotnet test` in the test/Chirp.ApiTests directory

Note: Ensure the BaseUrl in ApiTests.cs matches your local server address.

These tests provide a foundation for expanding our API testing suite as we continue to develop and refine the Chirp application.
